lib/web/terminal.go

```diff
@@ -853,6 +857,33 @@ func (t *TerminalHandler) streamEvents(tc *client.TeleportClient) {
 	}
 }
 
+// handleFileTransfer receives file transfer requests and responses and forwards them
+// to the SSh session
+func (t *TerminalHandler) handleFileTransfer(fileTransferC <-chan *session.FileTransferParams) {
+	for {
+		select {
+		case <-t.terminalContext.Done():
+			return
+		case transferRequest := <-fileTransferC:
+			// if not response, this is a new file transfer request
+			if !transferRequest.Response {
```

Comment:
Yeah, this grew from "eh only 1 field is needed" to a few, and I didn't cut them apart. It's best to do that now and make it more clear.

Reply:
After splitting the structs, I made a new `chan`, one to handle the requests, and one to handle the responses. The original reason to have this struct be the same was to reuse the same `chan` and just make decisions based on what info came through but I think thats a bit lazy now. Although I have a separate channel, I kept the "setting" of the channels in the same method (just with the two channels now instead of one) and also, have both cases in the same handler as well. Came out cleaner and easier to parse around throughout the app. Thanks for keeping me accountable